homeassistant/components/recorder/statistics.py

```diff
@@ -1256,6 +1256,23 @@ def _first_statistic(
     return None
 
 
+def _last_statistic(
+    session: Session,
+    table: type[StatisticsBase],
+    metadata_id: int,
+) -> datetime | None:
+    """Return the date of the newest statistic row for a given metadata id."""
+    stmt = lambda_stmt(
+        lambda: select(table.start_ts)
+        .filter(table.metadata_id == metadata_id)
+        .order_by(table.start_ts.desc())
+        .limit(1)
+    )
+    if stats := cast(Sequence[Row], execute_stmt_lambda_element(session, stmt)):
+        return dt_util.utc_from_timestamp(stats[0].start_ts)
+    return None
```

Comment:
I have some vague memory we can do it this in a simpler way.
There's the `ShortTermStatisticsRunCache`, could we use that instead?

Reply:
That would avoid looking index to find the newest `start_ts`, but its only for `StatisticsShortTerm` and it seems like we need `Statistics` here?